File: server/r/api/admin_api/get_admins.go
/*
 * Copyright (C) 2019 The Qing Project. All rights reserved.
 *
 * Use of this source code is governed by a license that can
 * be found in the LICENSE file.
 */

package adminapi

import (
	"net/http"
	"qing/app/appDB"
	"qing/app/appHandler"
	"qing/app/handler"
	"qing/da"
	"qing/r/rcom"
)

func getAdmins(w http.ResponseWriter, r *http.Request) handler.JSON {
	resp := appHandler.JSONResponse(w, r)

	admins, err := da.User.UnsafeSelectAdmins(appDB.Get().DB())
	if err != nil {
		panic(err)
	}
	userModels := make([]rcom.UserInfo, len(admins))
	for i, user := range admins {
		userModels[i] = rcom.NewUserInfo(user.ID, user.Name, user.IconName)
	}
	return resp.MustComplete(userModels)
}

File: it/api/admin/get_admins.js
/*
 * Copyright (C) 2021 The Qing Project. All rights reserved.
 *
 * Use of this source code is governed by a license that can
 * be found in the LICENSE file.
 */

import { post, user, assUtil, ass } from '../t.js';

post('get-admins: visitor', '/admin/get-admins', 0, null, (data) => {
  assUtil.notAuthorized(data);
});

post('get-admins: user', '/admin/get-admins', user.user, null, (data) => {
  assUtil.notAuthorized(data);
});

post('get-admins: admin', '/admin/get-admins', user.admin, null, (data) => {
  const adminData = data.data.find((d) => d.eid === user.admin.eid);
  ass.de(adminData, user.admin);
});
